src/server/StudioManagementSystem/Controllers/GroupApplicationController.cs

```diff
@@ -33,12 +33,26 @@ public ActionResult<bool> AddGroupInvitation(InvitationDto dto)
 
         return task.Result;
     }
+
+    [HttpPatch()]
+    public ActionResult<bool> RejectGroupApplication(RemoveInvitationDto dto)
+    {
+        var ct = _cancellationTokenAccessor.Token;
+        var task = _groupApplicationRepository.RejectGroupApplication(dto, ct);
+
+        task.Wait(ct);
+
+        if (!task.Result)
+            return StatusCode(500);
+
+        return task.Result;
+    }
     
     [HttpGet("{id}")]
     public List<GroupApplicationDto> GetGroupApplication(Guid id)
     {
         var ct = _cancellationTokenAccessor.Token;
-        var task = _groupApplicationRepository.GetGroupApplicationAsync(id, ct);
+        var task = _groupApplicationRepository.GetGroupApplicationsAsync(id, ct);
         task.Wait(ct);
 
```

Comment:
we should assert if the task was successful with a result before attempting to map and return,


```suggestion
if (!task.Result)
            return StatusCode(500);
```